Add Ruff

Mustard2 opened the pull request:
See #232

Mustard2 committed:
Create pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Update pylint.yml

Mustard2 committed:
Version 2025.5.0 (#250)

Mustard2 committed:
Version 2025.6.0 (#256)

Mustard2 committed:
Bump version to 2025.6.0 (#258)

Mustard2 committed:
MustardUI 2025.6.1 (#259)

Mustard2 committed:
Version 2025.6.2 (#260)

Mustard2 committed:
Version 2025.6.3 (#261)

Mustard2 committed:
Version 2025.6.4 (#263)

* Removed bugged Custom Properties option for Outfits. Now if custom properties are added to the outfits, they'll be always shown
* Fix for error when using restore morphs to default value for custom morphs

Mustard2 committed:
Version 2025.6.5 (#264)

-Fix: An error might occur if an Outfit is deleted or removed with the MustardUI buttons after the linked collection has been manually removed.
-Fix: An error might prevent custom properties to be enabled/disabled when an outfit is changed after a collection linked to an outfit has been manually removed.
-Fix: An error might occur when changing outfits if the bone physics is available.
-Fix: Fixed a wrong handling of status check in physics functions that was preventing the modifiers to be switched on/off correctly on the single physics item.

Mustard2 committed:
Version 2025.7 (#267)

- Feature: A new optimization option is available to disable heavy modifiers on Outfit switch, similarly to what is already happening with the Armature modifier. This option can be enabled in the Configuration mode, when Advanved Option is on.
- Enhancement: New Hair panel for particle hair. Instead of showing all possible systems on the object, a cleaner interface is now available to manage all the systems together.
- Fix: Fix an error that might occur when using the Rebind Physics option when some modifiers can not be switched on (for instance if the visibility option is driven by a driver and is not modifiable).
- Fix #269: Using the Clean tool might make the Outfit panel void.
- Fix #270: Hair and Extras custom properties were not cleaned using the Clean tool.

Mustard2 committed:
Version 2025.8 (#276)

- Feature #268: Possibility to add custom properties to the "Nude" outfit. It is also possible to flag to each "Nude" property to be switched on/off when changing to/from the "Nude" outfit.
- Feature: It is now possible to specify a minimum Blender version for the model. If the version condition is not met, a Warning will appear notifying the user about possible broken functionalities and suggesting to update Blender.
- Enhancement #275: Improved the Face Rig tool.
- Enhancement: New model versioning system.
- Enhancement: Bone Physics creator tool now works on disconnected chains, with the possibility to create a unique physics object or several ones for each disconnected bone chain.
- Enhancement #278 : Genesis 8 and Genesis 9 are now better renamed on import.
- Bug #277: If several instances of the same model are added in a file, the Outfit names are not properly rendered (in the cases where the names contain .xyz substrings where x,y,z are numbers).
- Bug: Using the Physics button near each outfit when several physics items were available for the same outfit piece, only one was correctly switched on/off.
- Bug: In Blender 5.0, the Physics Bone tool was not working (grayed out). Several errors might appear in the console about its poll function.

Mustard2 committed:
Version 2025.8.1 (#280)

- Feature: It is now possible to rename Outfit collections and their pieces, even if not added in the UI yet. An automatic process has been added to guess the possible outfit name. This should make the renaming of all collections and outfit pieces easier for the creators.
- Bug: Old system version was not shown in the Info & Settings panel.

Mustard2 committed:
Version 2025.8.2 (#282)

- Feature (#177): Hair can now be switched on/off automatically by showing/enabling single outfit pieces. This can be done by the new Hair Switcher Collection (https://files.catbox.moe/e6dcd2.mp4).
- Enhancement: The outfit/hair visibility operators (the ones called when switching outfits/hair, or single outfit pieces) have been completely rewritten. They are now considerably faster (up to the point where the simple hiding/showing Objects - as the default Blender operation - is the bottleneck).
- Bug (#279): Using the Smart Check button on Armatures always removes the current bone collections. Now an option has been introduced to override this behaviour if necessary.
- Bug (#244): Shift click on an Outfit piece with children object is not behaving as expected.

Mustard2 committed:
Version 2025.8.3 (#283)

* Bug: Fix for the Armature modifier not being activated on locked objects.

Mustard2 committed:
Version 2025.8.4 (#284)

Mustard2 committed:
Version 2025.8.5 (#287)

* Reintroduced support for Blender 4.2, 4.3, 4.4 and 4.4

Mustard2 committed:
Version 2026.1.0 (#288)

- Feature: New Morph preset panel. Presets can be used to save the values of Morphs, to be shared with others or to simplify the transfer to other models. The function takes care of checking if the Morphs are available: this means you can use presets on different models than the one where the preset was created, and only the Morphs available will be loaded neglecting the others.
- Enhancement: The Hair Physics button now is colored blue if any Hair physics is activated. This makes easier to understand if physics is enabled on any particle system without opening the whole menu.
- Enhancement: It is now possible to disable the Shape Keys and their drivers on the Outfits when they are not used (disabled by default, can be switched on in the Outfit panel of Configuration mode, when the Advanced options are active). In some tests I got +100% performance just enabling this new option.
- UI: The optimization settings of Outfits and Hair are not hidden by default anymore under the Advanced options. They now have a dedicated settings.
- Fix: Fixed the Morph default operator, which might fail in certain conditions. Also, the effects of the defaulted values is immediate, and does not require to move the camera or do another action on the software.

Mustard2 committed:
Version 2026.2.0 (#289)

- **Feature**: Added Creator Tool to link shape keys to active object. This can be useful when you have a main Object (e.g. the body) and you want to link other objects (e.g. outfits) shape keys that have the same name of the main one.
- **Feature**: Added Creator Tool to transfer single Vertex Groups from the active to the other selected Objects.
- **Feature**: The Morphs can now be Frozen to increase performance. This means that their shape keys and drivers will be disabled unless their values is different from 0. When Morphs are frozen, it is not possible to change their values.
- **Feature**: The Outfit removal tool is also available for users if Advanced Options is enabled.
- **Feature**: Support for single Armature Bone Collection selection (star `is_solo` operator). This should prevent confusion when these operators are active outside of the UI. The Reset Layer Visibility operator now also revert these star operators.
- **Feature**: Added a tool to transfer animation data from one armature to another.
- **Documentation**: The documentation has been improved for better readability. It has also been updated in several outdated parts.
- **Fix** (#272): Cleanup tool also remove dangling physics items.
- **Fix** (#291): Fixed an error that might occur when enabling the addon if some keymaps were not found.

Mustard2 committed:
Version 2026.2.1 (#292)

- UI: Improved the Physics items list in Configuration mode. To avoid confusion, the Cloth icon now means that an Outfit has been assigned to the item.
- Fix: The Outfit Deletion operator was not working correctly. It deleted the Outfit selected in Configuration mode, and not the one in the outfit list.
- Fix (partial #285): When deleting an Outfit, Mask status and Shape Key values were not reverted. Now all the masks of the deleted outfit are disabled, and the custom property values set to the default value before deletion.
- Fix: Fixed a warning when starting Blender.
- Fix: Boolean Custom Properties were rebuilt with the Rebuild Custom Properties tool as integers instead of booleans.

Mustard2 committed:
Bump Version 2026.2.1 (#295)

Mustard2 committed:
MustardUI Logo Change (#299)

Ckang3D committed:
Fix Mask modifier matching & Enhance Version Date configuration (#290)

* Refine mask modifier name matching to use endswith

Updated mask modifier checks in both ops_complete_outfit_visibility.py and ops_visibility.py to use mod.name.endswith(obj.name) instead of checking for substring presence.

This change ensures more precise matching of modifier names to object names.

* Add customizable date format for model version

Introduces options to select the date format for the model version, including new properties for date format and manual date entry. Updates the configuration logic to use the selected format and handle invalid dates gracefully. UI is updated to allow users to choose the date format and set the date vector if enabled.

* Apply suggestions from code review

* Update modifier name check to use endswith method

* Refactor outfit visibility operator in Blender

* Apply suggestions from code review

* Apply suggestion from @Mustard2

---------

Co-authored-by: Mustard <60780948+Mustard2@users.noreply.github.com>

Mustard2 committed:
Version 2026.3.0 (#297)

- Enhancement: The Simplify tool has been completely rewritten to support new features and solve some longstanding issues (#300 #304).
- Feature (#149 #187): Float subtype and step can now be set in the custom property settings.
- Enhancement (#290): The Version Date can now be overridden and customized with different date formats (thanks to @Ckang3D).
- Fix: An error with morph presets when using boolean values might occur.
- Fix (#290): Fixed an issue on the outfit Mask modifiers that might be enabled/disabled erroneously if outfits with names that are substrings of others are used (thanks to @Ckang3D).

Mustard2 committed:
Version 2026.3.1 (#305)

- Fix: Fix error when using the Custom Properties settings with Color properties.

Mustard2 committed:
Version 2026.3.2 (#308)

- Fix (#307): The UI button now also checks the presence of old Simplify settings, and adds the new recently introduced rewritten panel if it was already enabled in previous versions.

Mustard2 committed:
Version 2026.3.3 (#309)

- Fix: The Simplify tool was not working correctly with several armature in the scene.
- Fix: The Simplify tool might fail in some cases.

Mustard2 committed:
Remove extra blank line in pylint.yml

Mustard2 commented:
@cl3m3c7 Here I was using pylint, not ruff. Is it ok, or better to switch?

cl3m3c7 commented:
@Mustard2 not sure 😅

https://docs.astral.sh/ruff/faq/#how-does-ruffs-linter-compare-to-pylint

or both?

cl3m3c7 committed:
setup ruff instead of pylint

we will use ruff for now, if it does not help at linting, we will add it back later.

cl3m3c7 committed:
Format: global init

cl3m3c7 committed:
Format: armature module

cl3m3c7 committed:
Format: armature module

cl3m3c7 committed:
Merge pull request #313 from cl3m3c7/Dev-tools-addition

Mustard2 commented on __init__.py:
```diff
@@ -1,36 +1,41 @@
+    warnings,
+)
+
 # Add-on information
```
Putting bl_info after the import was creating issues in the past. Is it working correctly if installed as an addon AND as an extension now?

Mustard2 committed:
Merge branch 'main' into Dev-tools-addition

cl3m3c7 committed:
Format: armature module (cont.)

Mustard2 commented:
Sorry my bad, should be fine now I guess

cl3m3c7 committed:
Merge branch 'Mustard2:Dev-tools-addition' into Dev-tools-addition

Mustard2 commented:
I'll merge it after 2026.4.0 then. :)

cl3m3c7 committed:
Format: configuration module

cl3m3c7 committed:
Format: custom_properties module

cl3m3c7 committed:
Format: links module

cl3m3c7 committed:
add unused function param rule to ruff

cl3m3c7 commented:
I am working file by file basis, so it will take some time. go ahead ;)

cl3m3c7 committed:
Format: menu module

cl3m3c7 committed:
Format: misc package

cl3m3c7 committed:
Format: model_selection package

cl3m3c7 committed:
Format: morphs package

cl3m3c7 committed:
Format: outfits package

cl3m3c7 committed:
Format: physics package

cl3m3c7 committed:
Format: sections package

cl3m3c7 committed:
Format: settings package

cl3m3c7 committed:
Format: tools package

cl3m3c7 committed:
Format: tools_creators package

cl3m3c7 committed:
Format: viewport_panel package

cl3m3c7 committed:
Format: warnings package

cl3m3c7 committed:
Format: fix missed files

cl3m3c7 committed:
Ruff: disable unused param rule for now!

cl3m3c7 committed:
Merge pull request #314 from cl3m3c7/Dev-tools-addition

Refactor and format all the files based on Ruff

cl3m3c7 commented:
@Mustard2, seems I have finished before you, but I will wait until you merge #310, then I rebase, format + refactor and merge this PR.

cl3m3c7 committed:
Revert "Format: custom_properties module"

This reverts commit db0af8b049b4c39cd9067ea97ed1959e57b0e9ac.

cl3m3c7 committed:
Revert "Format: menu module"

This reverts commit ba8e82ca3824ca7cb7d765556de20a68c25f0f1f.

cl3m3c7 committed:
Format: custom_properties package

cl3m3c7 committed:
Format: menu package

cl3m3c7 committed:
Merge pull request #315 from cl3m3c7/Dev-tools-addition

Fix: addon stopped working after some refactors

Mustard2 commented:
Thank you very much for the huge work!

Probably it would have been easier to review if it was split into different MR, but I guess it should be fine.

I'll firstly merge 2026.4 so I have a version with the latest features to compare this one to. I can then try to use the UI a bit with this formatting to see if something breaks. Hopefully no, but it's too long to review in a standard way. :)

Mustard2 commented on armature/ops_transfer_animation.py:
```diff
@@ -2,10 +2,12 @@
 
 class MustardUI_Armature_TransferAnimation(bpy.types.Operator):
-    """Copy all animation data (actions and NLA strip settings) and world transform from one armature to the Active Object armature"""
+    """Copy all animation data (actions and NLA strip settings) and world transform from
```
This kind of comments unfortunately cannot be split into different lines (we need an exception). It's a Blender limitation.

Otherwise this happens, which is really ugly:

<img width="1062" height="200" alt="Image" src="https://github.com/user-attachments/assets/43b59e4e-8659-4554-ad17-89a9e3fe7e1e" />

<img width="468" height="155" alt="Image" src="https://github.com/user-attachments/assets/7d6dec1d-6a28-4f8b-86d5-d25a082d72fe" />

Mustard2 replied:
Or a solution is to use `bl_description` instead I guess

cl3m3c7 replied:
we can disable linting for class docstrings if that works!
anyways, I may not be able to connect for sometime, I will merge the latest main and address this, and If I didn't connect for some time (I don't know for how long), you may go and merge it!

and btw, there are some functions and methods that have unused params, but I disabled the rule for now (last comment in pyproject.toml) ! we may need to look for these later.

Mustard2 replied:
No worries, there's no hurry here. Thanks for all the help!

For the unused parameters, Blender bpy triggers some false positive. We might need to examine those carefully one by one to be sure that functionalities are not broken (I can do it).

Mustard2 commented on .github/workflows/lint.yml:
```diff
@@ -0,0 +1,27 @@
+name: Formatting and Linting
```
We should add an exception folder here from 2026.4, the diffeomorphic external files.

It's better not to touch them, I just want to update the files with a copy/paste.

cl3m3c7 replied:
I will exclude it.

Mustard2 replied:
Thanks!
